File: packages/ts-sdk/test/adapters/files.test.ts
import { describe, expect, it } from 'vitest';
import { FilesAdapter } from '../../src/treedx/index.js';
import { MockTransport } from './mock.js';

describe('FilesAdapter', () => {
  it('constructs file endpoints', async () => {
    const transport = new MockTransport();
    const adapter = new FilesAdapter({ transport });
    await adapter.tree('ws');
    await adapter.read('ws', { path: 'a.md' });
    await adapter.write('ws', {});
    await adapter.patch('ws', {});
    await adapter.delete('ws', { path: 'a.md' });
    await adapter.commit('ws', {});
    expect(transport.requests.map((r) => `${r.method} ${r.path}`)).toEqual([
      'GET /api/v1/workspaces/ws/tree',
      'GET /api/v1/workspaces/ws/files',
      'PUT /api/v1/workspaces/ws/files',
      'PATCH /api/v1/workspaces/ws/files',
      'DELETE /api/v1/workspaces/ws/files',
      'POST /api/v1/workspaces/ws/commit'
    ]);
  });
});

File: packages/elixir-sdk/test/adapters/files_test.exs
defmodule TreeDxSdk.FilesAdapterTest do
  use ExUnit.Case, async: true

  test "all constructs expected request" do
    {:ok, pid} = TreeDxSdk.Test.MockTransport.start_link()
    client = TreeDxSdk.Test.MockTransport.client(pid)
    TreeDxSdk.Files.tree(client, "ws/a")
    TreeDxSdk.Files.write(client, "ws/a", %{})
    TreeDxSdk.Files.patch(client, "ws/a", %{})
    TreeDxSdk.Files.delete(client, "ws/a")
    TreeDxSdk.Files.commit(client, "ws/a", %{})

    assert Enum.any?(
             TreeDxSdk.Test.MockTransport.requests(pid),
             &(&1.method == :post and &1.path == "/api/v1/workspaces/ws%2Fa/commit")
           )
  end
end
